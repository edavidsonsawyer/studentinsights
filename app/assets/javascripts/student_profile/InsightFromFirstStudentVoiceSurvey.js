import React from 'react';
import PropTypes from 'prop-types';
import {toMomentFromTimestamp} from '../helpers/toMoment';
import HelpBubble, {modalFromRightWithVerticalScroll} from '../components/HelpBubble';
import NoteText from '../components/NoteText';
import {fontSizeStyle} from './InsightQuote';
import InsightFromStudent from './InsightFromStudent';


// Render an insight from a student voice survey response
export default class InsightFromFirstStudentVoiceSurvey extends React.Component {
  render() {
    const {student, insightPayload} = this.props;
    const promptText = insightPayload.prompt_text;
    const responseText = insightPayload.survey_response_text;
    const completedSurvey = insightPayload.student_voice_completed_survey;
    const surveyMoment = toMomentFromTimestamp(completedSurvey.form_timestamp);

    return (
      <InsightFromStudent
        promptText={promptText}
        responseText={responseText}
        student={student}
        inWhatWhenEl={
          <div>
            <span>in </span>
            <HelpBubble
              style={{margin: 0}}
              modalStyle={modalFromRightWithVerticalScroll}
              linkStyle={fontSizeStyle}
              teaser="What I want my teachers to know"
              title="What I want my teachers to know"
              content={this.renderStudentSurveyDialog(completedSurvey)}
            />
            <span> on {surveyMoment.format('M/D/YY')}</span>
          </div>
        }
      />
    );
  }
  
  renderStudentSurveyDialog(completedSurvey) {
    return (
      <div style={{display: 'flex', flexDirection: 'column', overflowY: 'scroll'}}>
        <NoteText text={completedSurvey.survey_text} />
      </div>
    );
  }
}
InsightFromFirstStudentVoiceSurvey.propTypes = {
  student: PropTypes.shape({
    id: PropTypes.number.isRequired,
    first_name: PropTypes.string.isRequired,
    last_name: PropTypes.string.isRequired
  }).isRequired,
  insightPayload: PropTypes.shape({
    prompt_key: PropTypes.string.isRequired,
    prompt_text: PropTypes.string.isRequired,
    survey_response_text: PropTypes.string.isRequired,
    student_voice_completed_survey: PropTypes.shape({
      form_timestamp: PropTypes.string.isRequired,
      survey_text: PropTypes.string.isRequired
    }).isRequired
  })
};


export const FROM_FIRST_STUDENT_VOICE_SURVEY = 'from_first_student_voice_survey';
