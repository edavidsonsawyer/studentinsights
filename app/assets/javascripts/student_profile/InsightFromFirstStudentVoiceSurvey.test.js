import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import InsightFromFirstStudentVoiceSurvey from './InsightFromFirstStudentVoiceSurvey';

function testProps(props) {
  return {
    insightPayload: {
      prompt_key: 'proud',
      prompt_text: 'What are you most proud of?',
      survey_response_text: 'Doing well in English and starting the recyling club.',
      student_voice_completed_survey: {
        form_timestamp: '2017-12-16T00:00:00.000Z',
        survey_text: '<full survey text>'
      }
    },
    student: {
      id: 42,
      first_name: 'Mari',
      last_name: 'Kenobi'
    },
    ...props
  };
}

// since element sizing methods aren't present in the Jest JS environment
jest.mock('../components/FitText', () => 'mocked-fit-text');

it('renders without crashing', () => {
  const el = document.createElement('div');
  ReactDOM.render(<InsightFromFirstStudentVoiceSurvey {...testProps()} />, el);
});

it('snapshots view', () => {
  const tree = renderer
    .create(<InsightFromFirstStudentVoiceSurvey {...testProps()} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});