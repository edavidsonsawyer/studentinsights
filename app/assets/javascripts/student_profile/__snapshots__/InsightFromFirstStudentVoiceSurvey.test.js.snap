// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots view 1`] = `
<div
  className="InsightQuote InsightFromStudent "
  style={
    Object {
      "display": "flex",
      "flex": 1,
      "flexDirection": "column",
    }
  }
>
  <div
    style={
      Object {
        "display": "flex",
        "flex": 1,
        "margin": 20,
        "marginBottom": 0,
        "marginTop": 15,
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
          "flex": 1,
          "flexDirection": "column",
          "overflow": "hidden",
        }
      }
    >
      <div
        style={
          Object {
            "display": "flex",
            "flex": 1,
            "flexDirection": "column",
          }
        }
      >
        <div
          style={
            Object {
              "fontSize": 12,
            }
          }
        >
          What are you most proud of?
        </div>
        <div
          style={
            Object {
              "display": "flex",
              "flex": 1,
              "flexDirection": "column",
              "marginTop": 5,
            }
          }
        >
          <mocked-fit-text
            fontSizeStep={4}
            maxFontSize={48}
            minFontSize={12}
            text="“Doing well in English and starting the recyling club.”"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "alignItems": "flex-end",
        "display": "flex",
        "flexDirection": "row",
        "fontSize": 12,
        "justifyContent": "space-between",
        "margin": 20,
        "marginBottom": 15,
        "marginTop": 10,
      }
    }
  >
    <div>
      <div>
        <div>
          from 
          <a
            href="/students/42"
            rel="noopener noreferrer"
            style={
              Object {
                "fontSize": 12,
              }
            }
            target="_blank"
          >
            Mari
             
            Kenobi
          </a>
        </div>
        <div>
          <div>
            <span>
              in 
            </span>
            <div
              style={
                Object {
                  "display": "inline",
                  "margin": 0,
                  "marginLeft": 10,
                }
              }
            >
              <a
                href="#"
                onClick={[Function]}
                style={
                  Object {
                    "fontSize": 12,
                    "outline": "none",
                  }
                }
              >
                What I want my teachers to know
              </a>
            </div>
            <span>
               on 
              12/16/17
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
